py/kubeflow/testing/go-license-tools/README.md

```diff
@@ -6,41 +6,137 @@ When we release go library images (can be considered as redistributing third
 party binary).
 
 We need to do the following to be compliant:
-* Put license declarations in the image for licences of all dependencies and transistive dependencies.
+* Put license declarations in the image for licenses of all dependencies and transitive dependencies.
 * Mirror source code in the image for code with MPL, EPL, GPL or CDDL licenses.
 
 It's not an easy task to get license of all (transitive) dependencies of a go
 library. Thus, we need these tools to automate this task.
 
 ## How to get all dependencies with license and source code?
 
-1. Install CLI tools here: `python setup.py install`
-1. Collect dependencies + transitive dependencies in a go library. Put them together in a text file called `dep.txt`. Format: each line has a library name. The library name should be a valid golang import module name.
+### I. Setup
+Download this folder to your local folder namely `<license_tool>` and install it:
+```
+$ python <license_tool>/setup.py install
+```
 
-    Example ways to get it:
-    * argo uses gopkg for package management. It has a [Gopkg.lock file](https://github.com/argoproj/argo/blob/master/Gopkg.lock)
-    with all of its dependencies and transitive dependencies. All the name fields in this file is what we need. You can run `parse-toml-dep` to parse it.
-    * minio uses [official go modules](https://blog.golang.org/using-go-modules), there's a [go.mod file](https://github.com/minio/minio/blob/master/go.mod) describing its direct dependencies. Run command `go list -m all` to get final versions that will be used in a build for all direct and indirect dependencies, [reference](https://github.com/golang/go/wiki/Modules#daily-workflow). Parse its output to make a file we need.
+### II. Get all dependency repositories
+1. Collect dependencies and transitive dependencies in a Go library into a text file called `dep.txt`, where each line is a valid golang import module name. For example
+    ```
+    ......
+    cloud.google.com/go
+    github.com/BurntSushi/toml
+    github.com/beorn7/perks
+    github.com/bmatcuk/doublestar
+    ......
+    ```
 
-    Reminder: don't forget to put the library itself into `dep.txt`.
```

Comment:
Shall we keep this reminder?
For redistributing 3rd party images, we need the library itself's license.
For distributing first party images, I think it's better we deliver our own license too.

Reply:
Done. Added to line 43

Reply:
Done.